src/test/kotlin/uk/gov/dluhc/printapi/mapper/PrintDetailsMapperTest.kt

```diff
@@ -0,0 +1,135 @@
+
+        val expected = with(message) {
+            PrintDetails(
+                id = UUID.randomUUID(),
+                requestId = getAMongoDbId(),
+                sourceReference = sourceReference,
+                applicationReference = applicationReference,
+                sourceType = SourceTypeEntity.VOTER_CARD,
+                vacNumber = randomAlphanumeric(20),
+                requestDateTime = requestDateTime,
+                firstName = firstName,
+                middleNames = middleNames,
+                surname = surname,
+                certificateLanguage = certificateLanguageEntity,
+                photoLocation = photoLocation,
+                delivery = with(delivery) {
+                    CertificateDelivery(
+                        addressee = addressee,
+                        address = with(address) {
+                            Address(
+                                street = street,
+                                postcode = postcode,
+                                property = property,
+                                locality = locality,
+                                town = town,
+                                area = area,
+                                uprn = uprn
+                            )
+                        },
+                        deliveryClass = DeliveryClass.STANDARD
+                    )
+                },
+                gssCode = gssCode,
+                issuingAuthority = localAuthority.name,
+                issueDate = LocalDate.now(),
+                eroEnglish = with(ero) {
+                    ElectoralRegistrationOffice(
+                        name = name,
+                        phoneNumber = null,
+                        emailAddress = null,
+                        website = null,
+                        address = null
+                    )
+                },
+                eroWelsh = if (certificateLanguageModel == CertificateLanguage.EN) null else with(ero) {
+                    ElectoralRegistrationOffice(
+                        name = name,
+                        phoneNumber = null,
+                        emailAddress = null,
+                        website = null,
+                        address = null
+                    )
+                }
+            )
+        }
+        // When
+
+        val actual = mapper.toPrintDetails(message, ero, localAuthority.name)
+
+        // Then
+        assertThat(actual).usingRecursiveComparison().ignoringFields("id", "requestId", "vacNumber").isEqualTo(expected)
+        assertThat(actual.id).isNotNull
+        assertThat(actual.requestId).isNotNull.containsPattern(Regex("^[a-f\\d]{24}$").toPattern())
+        assertThat(actual.vacNumber).isNotNull.containsPattern(Regex("^[A-Za-z\\d]{20}$").toPattern())
+        verify(sourceTypeMapper).toSourceTypeEntity(SourceTypeModel.VOTER_MINUS_CARD)
+    }
+
+    @Test
+    fun `should map id requestId and vacNumber to random values`() {
```

Comment:
Is this test testing functionality of the mapper of the `IdFactory` ?
As per my comment above I think these values should be coming from the `IdFactory`, and therefore this test is not required?
What do you think?

Reply:
Thanks, for pointing this @NathanRussellValtech. 
I have removed this test and just assert the `IdFactory` is used generating requestId and vacNumber in the test above it.